pkg/updater/resultstore/resultstore.go

```diff
@@ -458,6 +480,110 @@ func processGroup(tg *configpb.TestGroup, group *invocationGroup) *updater.Infla
 	return col
 }
 
+// filterProperties returns the subset of results containing all the specified properties.
+func filterProperties(results []*resultstorepb.Test, properties []*configpb.TestGroup_KeyValue) []*resultstorepb.Test {
+	if len(properties) == 0 {
+		return results
+	}
+
+	var out []*resultstorepb.Test
+
+	match := make(map[string]bool, len(properties))
+
+	for _, p := range properties {
+		match[p.Key] = true
+	}
+
+	for _, r := range results {
+		found := map[string]string{}
+		fillProperties(found, r, match)
+		var miss bool
+		for _, p := range properties {
+			if found[p.Key] != p.Value {
+				miss = true
+				break
+			}
+		}
+		if miss {
+			continue
+		}
+		out = append(out, r)
+	}
+	return out
+}
+
+// fillProperties reduces the appendProperties result to the single value for each key or "*" if a key has multiple values.
+func fillProperties(properties map[string]string, result *resultstorepb.Test, match map[string]bool) {
+	if result == nil {
+		return
+	}
+	multiProps := map[string][]string{}
+
+	appendProperties(multiProps, result, match)
+
+	for key, values := range multiProps {
+		if len(values) > 1 {
+			var diff bool
+			for _, v := range values {
+				if v != values[0] {
+					properties[key] = "*"
+					diff = true
+					break
+				}
+			}
+			if diff {
+				continue
+			}
+		}
+		properties[key] = values[0]
+	}
+}
+
+// appendProperties from result and its children into a map, optionally filtering to specific matching keys.
+func appendProperties(properties map[string][]string, result *resultstorepb.Test, match map[string]bool) {
```

Comment:
Think it's worth adding some simple test cases for this.

Reply:
I think we can skip this since `fillProperties` calls this and already include tests there to cover the code and logic here